vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -284,49 +323,53 @@ func (d *VolumeDriver) Create(r volume.Request) volume.Response {
 		// in which case we continue creating the file system.
 		fs.DevAttachWait(watcher, r.Name, device)
 	}
-	errMkfs := fs.Mkfs(mkfscmd, r.Name, device)
-	if errMkfs != nil {
-		log.WithFields(log.Fields{"name": r.Name,
-			"error": errMkfs}).Error("Create filesystem failed, removing the volume ")
-		errDetach := d.ops.Detach(r.Name, nil)
-		if errDetach != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": errDetach}).Warning("Detach volume failed ")
+	if err := fs.Mkfs(mkfscmd, r.Name, device); err != nil {
+		log.WithField("name", r.Name).WithError(err).Error(
+			"Create filesystem failed, removing the volume")
+		if err := d.ops.Detach(r.Name, nil); err != nil {
+			log.WithField("name", r.Name).WithError(err).Warn(
+				"Detach volume failed")
 		}
-		errRemove := d.ops.Remove(r.Name, nil)
-		if errRemove != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": errRemove}).Warning("Remove volume failed ")
+		if err := d.ops.Remove(r.Name, nil); err != nil {
+			log.WithField("name", r.Name).WithError(err).Warn(
+				"Remove volume failed")
 		}
-		return volume.Response{Err: errMkfs.Error()}
+		return volume.Response{Err: err.Error()}
```

Comment:
Hi @govint,

I've closed this PR, but I wanted to let you know that the original errors *are* maintained. The line:

`if err := d.ops.Remove(r.Name, nil); err != nil {`

Does not override the `err` object above it:

`if err := fs.Mkfs(mkfscmd, r.Name, device); err != nil {`

This is due to Go scoping. Here's a quick example in the Go playground ([run](https://play.golang.org/p/f02pmxhXP9)):

```go
package main

import (
	"fmt"
)

func main() {
	x := 0
	if x := 2 + 3; x > 0 {
		fmt.Printf("x=%d\n", x)
	}
	fmt.Printf("x=%d\n", x)
}
```

The above code will print the following:

```
x=5
x=0
```

This is because the `x` in the `if` block is not the same `x` in the main program. It's a new scope.